src/Client/BuzzHouse/Generator/ExternalIntegrations.cpp

```diff
@@ -1012,28 +1012,29 @@ bool MinIOIntegration::sendRequest(const String & resource)
     hints.ai_socktype = SOCK_STREAM;
     hints.ai_protocol = IPPROTO_TCP;
 
+    /// It should always fit, ignore the result
     auto u = std::sprintf(buffer, "%" PRIu32 "", sc.port);
     UNUSED(u);
     if ((error = getaddrinfo(sc.hostname.c_str(), buffer, &hints, &result)) != 0)
     {
         if (error == EAI_SYSTEM)
         {
-            strerror_r(errno, buffer, sizeof(buffer));
-            LOG_ERROR(fc.log, "getnameinfo error: {}", buffer);
+            LOG_ERROR(fc.log, "getnameinfo error: {}", strerror(errno));
         }
         else
         {
             LOG_ERROR(fc.log, "getnameinfo error: {}", gai_strerror(error));
         }
         return false;
     }
+    SCOPE_EXIT({ close(sock); });
```

Comment:
If you want al sockets to be closed, you should move this within the for-loop. Before L1035:

```cpp
SCOPE_EXIT({ if (sock) close(sock); });
```